WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -183,26 +183,32 @@ def __setJobStatusBulk(cls, jobID, statusDict):
     lastTime = Time.toString(Time.fromEpoch(lastTime))
 
     dates = sorted(statusDict)
+    # If real updates, start from the current status
+    if dates[0] >= lastTime and not status:
+      status = currentStatus
     log = gLogger.getSubLogger('JobStatusBulk/Job-%s' % jobID)
     log.debug("*** New call ***", "Last update time %s - Sorted new times %s" % (lastTime, dates))
-    # Remove useless items in order to make it simpler later, although should not be there
+    # Remove useless items in order to make it simpler later, although there should not be any
     for sDict in statusDict.values():
```

Comment:
This, combined with `pop` below, would not survive in python 3 (`values()` returns an iterator, and would raise `RuntimeError: dictionary changed size during iteration`.

Reply:
No, the `.pop()` if for within each value ;-) 
`sDict` is not removed, only its content, this is why I had to put either a `list(sDict)` or `sorted(sDict)`